### PR TITLE
Change TelemetryHelper to non static class

### PR DESCRIPTION
**Pull request by rita-sonarsource**
Prep for #2997

### CODE REVIEW

File: src/Integration/Telemetry/TelemetrySerializer.cs [hunk 1]
```diff
@@ -0,0 +1,32 @@
...
+ * You should have received a copy of the GNU Lesser General Public License
+ * along with this program; if not, write to the Free Software Foundation,
+ * Inc., 51 Franklin Street, Fifth Floor, Boston, MA  02110-1301, USA.
+ */
+
+using SonarQube.Client.Helpers;
+
+namespace SonarLint.VisualStudio.Integration
+{
+    public static class TelemetrySerializer
```
Comment (rita-sonarsource) [inline src/Integration/Telemetry/TelemetrySerializer.cs:25]: Extracted as-is from "TelemetryHelper" and renamed

File: src/Integration/Telemetry/TelemetryPayloadCreator.cs [hunk 2]
```diff
@@ -19,35 +19,72 @@
...
 {
-    public static class TelemetryHelper
+    public interface ITelemetryPayloadCreator
     {
+        TelemetryPayload Create(TelemetryData telemetryData);
+    }
+
+    [Export(typeof(ITelemetryPayloadCreator))]
+    [PartCreationPolicy(CreationPolicy.Shared)]
+    public class TelemetryPayloadCreator : ITelemetryPayloadCreator
```
Comment (rita-sonarsource) [inline src/Integration/Telemetry/TelemetryPayloadCreator.cs:38]: MEF exportable wrapper around "TelemetryHelper". Extracted as is.

File: src/Integration.UnitTests/Telemetry/TelemetryManagerTests.cs [hunk 3]
```diff
@@ -619,26 +607,6 @@ public void TaintIssueInvestigatedRemotely_CounterIncremented(int previousCounte
             telemetryRepositoryMock.Verify(x => x.Save(), Times.Once);
         }
 
-        [TestMethod]
-        [DataRow(true)]
-        [DataRow(false)]
-        public void SendPayload_SendsVsVersion(bool isVersionNull)
```
Comment (rita-sonarsource) [inline src/Integration.UnitTests/Telemetry/TelemetryManagerTests.cs:625]: No longer relevant as it's calculated inside the payload class

File: src/Integration.UnitTests/Telemetry/TelemetryManagerTests.cs [hunk 4]
```diff
@@ -660,14 +628,19 @@ public void SendPayload_DisabledSecretRulesAreUpdated()
...
 
-            TriggerTimerElapsed();
+            var payload = new TelemetryPayload();
 
-            telemetryClientMock.Verify(x => x.SendPayloadAsync(
-                It.Is((TelemetryPayload payload) => 
-                    payload.RulesUsage.EnabledByDefaultThatWereDisabled.Count == 2 &&
-                    payload.RulesUsage.EnabledByDefaultThatWereDisabled.Contains("secrets:rule3") &&
-                    payload.RulesUsage.EnabledByDefaultThatWereDisabled.Contains("secrets:rule6"))), Times.Once);
+            payloadCreatorMock.Setup(x => x.Create(
```
Comment (rita-sonarsource) [inline src/Integration.UnitTests/Telemetry/TelemetryManagerTests.cs:633]: Small tweak, same logic as before except I'm verifying that it's passed correctly to the payload class instead the client

**Review by duncanp-sonar — APPROVED** [review]

**Review by ugras-ergun-sonarsource — APPROVED** [review]

**Pull request merged**